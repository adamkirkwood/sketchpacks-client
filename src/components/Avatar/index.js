import React from 'react'
import LazyLoad from 'react-lazyload'

const Avatar = ({ handle, width, height }) => (
  <div>
    <LazyLoad height={height} offset={200} unmountIfInvisible debounce={500}>
      <img className="image is-maker"
        src={`https://github.com/${handle}.png`}
        role="presentation"
        width={width}
        height={height}
      />
    </LazyLoad>
  </div>
)

Avatar.defaultProps = { width: 24, height: 24 }

export default Avatar
