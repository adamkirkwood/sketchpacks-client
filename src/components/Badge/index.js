import React from 'react'

import './styles.scss'

const Badge = ({ value }) => (
  <span className={'badge'}>{value}</span>
)

Badge.defaultProps = { value: '' }

export default Badge
