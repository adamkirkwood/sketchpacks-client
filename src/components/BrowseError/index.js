import React from 'react'

const BrowseError = ({ message, onRetry }) => (
  <div className="empty-state">
    <p>ðŸ˜³ Could not fetch plugins. { message }</p>
    <button onClick={onRetry}>Retry</button>
  </div>
)

export default BrowseError
