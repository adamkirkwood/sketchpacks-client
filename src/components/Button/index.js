import React from 'react'

const Button = ({ actionVerb, onClick, className }) => (
  <button className={className} onClick={onClick}>{actionVerb}</button>
)

export default Button
