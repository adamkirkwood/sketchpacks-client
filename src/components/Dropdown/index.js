import React, { Component } from 'react'

import semver from 'semver'
import { setVersionRange,getPatchLevelLock,getMinorLevelLock } from 'reducers/sketchpack'
import { sanitizeSemVer } from 'lib/utils'

import LockGauge from 'components/LockGauge'

import 'normalize.css'
import 'css/milligram.scss'

class Dropdown extends Component {
  constructor (props) {
    super(props)

    this.state = {
      visible: false,
      lock_gauge: 'unlocked'
    }

    this.toggleVisibility = this.toggleVisibility.bind(this)

    this.handleFullLock = this.handleFullLock.bind(this)
    this.handlePatchLock = this.handlePatchLock.bind(this)
    this.handleMinorLock = this.handleMinorLock.bind(this)
    this.handleUnlocked = this.handleUnlocked.bind(this)
  }

  toggleVisibility () {
    this.setState({
      visible: !this.state.visible
    })
  }

  handleFullLock () {
    const {installed_version, identifier} = this.props
    this.props.handlePluginEvent({
      type: 'lock',
      version: installed_version,
      identifier: identifier,
      lock_strength: 'full'
    })
    this.toggleVisibility()
    this.setState({ lock_gauge: 'full' })
  }

  handleMinorLock () {
    const {installed_version, identifier} = this.props
    this.props.handlePluginEvent({
      type: 'lock',
      version: installed_version,
      identifier: identifier,
      lock_strength: 'minor'
    })
    this.toggleVisibility()
    this.setState({ lock_gauge: 'minor' })
  }

  handlePatchLock () {
    const {installed_version, identifier} = this.props
    this.props.handlePluginEvent({
      type: 'lock',
      version: installed_version,
      identifier: identifier,
      lock_strength: 'patch'
    })
    this.toggleVisibility()
    this.setState({ lock_gauge: 'patch' })
  }

  handleUnlocked () {
    const {installed_version, identifier} = this.props
    this.props.handlePluginEvent({
      type: 'lock',
      version: installed_version,
      identifier: identifier,
      lock_strength: 'unlocked'
    })
    this.toggleVisibility()
    this.setState({ lock_gauge: 'unlocked' })
  }

  render () {
    return (

      <div className="lock-select">
        <div className="lock-viewer button--dropdown" onClick={this.toggleVisibility}>
          <LockGauge gauge={this.state.lock_gauge} />

          <span className="lock__label">v{this.props.version}</span>
        </div>

        <ul className={this.state.visible
            ? 'dropdown__menu is-visible'
            : 'dropdown__menu'}>
          <li onClick={this.handleUnlocked}>
            <div className="lock-select__option">
              <strong>Unlocked</strong>
              <p>Apply all updates automatically</p>
            </div>
          </li>

          <li onClick={this.handleMinorLock}>
            <div className="lock-select__option">
              <strong>Minor Updates</strong>
              <p>Apply all updates less than v{sanitizeSemVer(semver.toComparators(getMinorLevelLock(this.props.version))[0][1])}</p>
            </div>
          </li>

          <li onClick={this.handlePatchLock}>
            <div className="lock-select__option">
              <strong>Patch Updates</strong>
              <p>Apply all updates less than v{sanitizeSemVer(semver.toComparators(getPatchLevelLock(this.props.version))[0][1])}</p>
            </div>
          </li>

          <li onClick={this.handleFullLock}>
            <div className="lock-select__option">
              <strong>Full Lock</strong>
              <p>Lock at v{this.props.version}</p>
            </div>
          </li>
        </ul>
      </div>

    )
  }
}


export default Dropdown
