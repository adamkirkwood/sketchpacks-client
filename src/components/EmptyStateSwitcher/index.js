import React from 'react'

const EmptyManagedState = () => {
  return (
    <div className="empty-state empty-state--expanded">
      <span className="o-emoji o-emoji--xl">😐</span>
      <h4>No installed plugins</h4>

      <p>Managed plugins will appear here.</p>
    </div>
  )
}

const EmptyUnmanagedState = () => {
  return (
    <div className="empty-state empty-state--expanded">
      <span className="o-emoji o-emoji--xl">😐</span>
      <h4>No unmanaged plugins installed</h4>

      <p>Unidentified plugins will appear here.</p>
    </div>
  )
}

const EmptyUpdatesState = () => {
  return (
    <div className="empty-state empty-state--expanded">
      <span className="o-emoji o-emoji--xl">🖖🏽</span>

      <h4>No updates available</h4>

      <p>Available plugin updates will appear here.</p>
    </div>
  )
}

const EmptyBrowseState = () => {
  return (
    <div className="empty-state empty-state--expanded">
      <span className="o-emoji o-emoji--xl">⚡️</span>
      <h4>Fetching plugins</h4>
    </div>
  )
}

const EmptySearchResultsState = () => {
  return (
    <div className="empty-state empty-state--expanded">
      <span className="o-emoji o-emoji--xl">🔭</span>
      <h4>No results found</h4>
    </div>
  )
}


const STATES = {
  managed: EmptyManagedState,
  unmanaged: EmptyUnmanagedState,
  updates: EmptyUpdatesState,
  browse: EmptyBrowseState,
  search: EmptySearchResultsState
}

const EmptyStateSwitcher = ({ pathname }) => {
  const segments = pathname.split('/')
  const ctx = segments[segments.length-1]

  const EmptyState = STATES[ctx] || STATES['browse']

  return <EmptyState />
}

export default EmptyStateSwitcher
