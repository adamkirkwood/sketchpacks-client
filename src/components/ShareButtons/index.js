import React from 'react'

// Facebook Like Button
export const LikeButton = ({ href, layout, action, size, showFaces, share }) => (
  <div
    className="fb-like"
    data-href={href}
    data-layout={layout}
    data-action={action}
    data-size={size}
    data-show-faces={showFaces}
    data-share={share}>
  </div>
)

LikeButton.propTypes = {
  href: React.PropTypes.string,
  layout: React.PropTypes.oneOf(['standard', 'button_count', 'button', 'box_count']),
  show_faces: React.PropTypes.bool,
  action: React.PropTypes.oneOf(['like', 'recommend']),
  size: React.PropTypes.oneOf(['small', 'large']),
  width: React.PropTypes.number,
  share: React.PropTypes.bool
}

// Twitter Tweet Button
export const TweetButton = ({ text, url, via, related, showCount }) => (
  <a href="https://twitter.com/share"
    className="twitter-share-button"
    data-text={text}
    data-url={url}
    data-via={via}
    data-related={related}
    data-show-count={showCount}
  >
    Tweet
  </a>
)

TweetButton.propTypes = {
  text: React.PropTypes.string,
  url: React.PropTypes.string,
  via: React.PropTypes.string,
  related: React.PropTypes.string,
  showCount: React.PropTypes.bool,
}
