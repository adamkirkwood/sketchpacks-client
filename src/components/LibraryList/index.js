import React, {Component} from 'react'

import LibraryMedia from 'components/LibraryMedia'
import BrowseError from 'components/BrowseError'
import EmptyStateSwitcher from 'components/EmptyStateSwitcher'

import {
  getPluginByIdentifier,
  selectPluginBasics
} from 'reducers'

import './styles.scss'

class LibraryList extends Component {
  constructor (props) {
    super(props)
  }

  render () {
    const {
      plugins,
      location,
      dispatch,
      handlePluginEvent,
      state,
      errorMessage,
      onRetry
    } = this.props

    if (errorMessage && plugins.length === 0) {
      return <BrowseError
        message={errorMessage}
        onRetry={onRetry} />
    }

    if (plugins.length === 0) {
      return <EmptyStateSwitcher pathname={this.props.location.pathname} />
    }

    return (
      <div className="o-plugin-list">

        {plugins.map((identifier, idx) => {
          return <LibraryMedia
            key={`${idx}-${identifier}`}
            plugin={selectPluginBasics(state,identifier)}
            location={location}
            dispatch={dispatch}
            handlePluginEvent={handlePluginEvent} />
        })}
      </div>
    )
  }
}

export default LibraryList
