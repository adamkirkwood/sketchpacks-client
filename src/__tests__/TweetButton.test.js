import React from 'react'
import {TweetButton} from '../components/ShareButtons'

describe('TweetButton', () => {
  it('should render a Twitter Tweet button', () => {
    const wrapper = shallow(
      <TweetButton url="https://sketchpacks.com"
        text="Sketchpacks for macOS â€” Manage your Sketch plugins from the menu bar"
        via="sketchpacks"
        related="adamkirkwood"
        showCount />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
