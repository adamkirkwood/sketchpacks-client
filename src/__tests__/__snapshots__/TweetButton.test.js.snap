// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetButton should render a Twitter Tweet button 1`] = `
<a
  className="twitter-share-button"
  data-related="adamkirkwood"
  data-show-count={true}
  data-text="Sketchpacks for macOS â€” Manage your Sketch plugins from the menu bar"
  data-url="https://sketchpacks.com"
  data-via="sketchpacks"
  href="https://twitter.com/share"
>
  Tweet
</a>
`;
